src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -404,32 +427,37 @@ function genFragment(
     lastList = nodeName;
   }
 
+  const blockType = getBlockTypeForTag(nodeName, lastList, blockRenderMap);
+
   // Block Tags
-  if (!inBlock && blockTags.indexOf(nodeName) !== -1) {
-    chunk = getBlockDividerChunk(
-      getBlockTypeForTag(nodeName, lastList, blockRenderMap),
-      depth,
-    );
+  if (lastList && inBlock === 'li' && nodeName === 'li') {
+    chunk = getBlockDividerChunk(blockType, depth, parentKey);
+    blockKey = chunk.blocks[0].key;
     inBlock = nodeName;
-    newBlock = true;
-  } else if (lastList && inBlock === 'li' && nodeName === 'li') {
+    nextBlockType =
+      lastList === 'ul' ? 'unordered-list-item' : 'ordered-list-item';
+    newBlock = !experimentalTreeDataSupport;
```

Comment:
this flag `newBlock` was previously used to split nested blocks, when we have tree support enabled we dont want this to be active